### PR TITLE
Handle side-effects

### PR DESCRIPTION
**Pull request by mrm007**
This PR updates the [`sideEffects` flag][flag] if needed. The work is needed for [Compiled Braid](https://github.com/seek-oss/braid-design-system/pull/1202).

If an entry point has side-effects (as defined via **package.json**'s `"sideEffects"` key) Crackle will update the `"sideEffects"` key so it also matches the output path in `dist`.

[flag]: https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free

Based on what's in `src` and the value of the flag, when running `crackle package` these things will happen:
- `.css.ts` files (and files importing them) will be placed in `dist/styles/<same-path-as-in-src>`
- files with side-effects (matching the globs defined in `"sideEffects"`) will be placed in `dist/side-effects/<same-path-as-in-src>`
- if an entry has side-effects the `"sideEffects"` key will be updated so it also matches the output path in `dist`.

To test, I've added a fixture `with-side-effects` that is a stripped-down version of Braid, as well as all of `braid-design-system` monorepo as a Git submodule. ~~I'll add instructions how to sparsely clone just the `braid-design-system` package, but everything works even without checking out the Git submodule.~~
I've added a script that selectively checks out the `braid-design-system` package from the monorepo.

### CODE REVIEW

File: scripts/braid-fixture.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+set -eox pipefail
+
+repo=git@github.com:seek-oss/braid-design-system.git
```
Comment (mrm007) [inline scripts/braid-fixture.sh:4]: This script selectively checks out the `braid-design-system` package from the monorepo.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[submodule "fixtures/braid-design-system"]
+	path = fixtures/braid-design-system
+	url = git@github.com:seek-oss/braid-design-system.git
+	branch = compiled-braid
```
[the remark this answers is not in the document]
Comment (mrm007) [reply .gitmodules:4]: It's useful for quick testing while we work on compiled Braid. Once v32 is out I agree we should remove the submodule.

File: packages/core/src/constants.ts [hunk 3]
```diff
@@ -1 +1,11 @@
 export const clientEntry = require.resolve('../entries/client.tsx');
+
+export const srcDir = 'src';
+export const distDir = 'dist';
```
[the remark this answers is not in the document]
Comment (mrm007) [reply packages/core/src/constants.ts:4]: This one's for packages (`crackle package`), the other one is for the site build (`crackle build`). I'll add a few comments.

File: packages/core/src/package/bundle.ts [hunk 4]
```diff
@@ -46,38 +57,36 @@ export const createBundle = async (
...
+              !srcPath.startsWith('.'),
+              `relative path ${srcPath} should not start with '.'`,
+            );
+
             if (
-              cssFileFilter.test(id) ||
-              getModuleInfo(id)?.importers.some((importer) =>
-                cssFileFilter.test(importer),
-              )
+              isVanillaFile(id) ||
```
[the remark this answers is not in the document]
Comment (mrm007) [reply packages/core/src/package/bundle.ts:78]: At the moment we are pretty coupled with vanilla-extract (see [here][1] and [here][2]), but I agree it makes sense to revisit that later.

[1]: https://github.com/seek-oss/crackle/blob/4e831559a5d6107fa4226e854ecfa627e7cd1b58/packages/core/src/plugins/rollup/vanilla-extract-debug-ids.ts
[2]: https://github.com/seek-oss/crackle/blob/4ea357faf06affe880741581689508a3b3b348e6/packages/core/src/plugins/esbuild/fix-vite-vanilla-extract-dep-scan.ts

File: packages/core/src/package/bundle.ts [hunk 5]
```diff
@@ -46,38 +57,36 @@ export const createBundle = async (
...
+              getModuleInfo(id)?.importers.some(isVanillaFile)
             ) {
-              const [_projectRoot, rawLocalPath] = id.split('src/');
-              const localPath = rawLocalPath.replace('/', '-');
+              // TODO: check if bundling all styles works
```
[the remark this answers is not in the document]
Comment (mrm007) [reply packages/core/src/package/bundle.ts:81]: I did a quick test and to me it looked like it was working. @benjervis was sceptical so I left it as a TODO for later.

[the remark this answers is not in the document]
Comment (michaeltaranto) [reply packages/core/src/package/bundle.ts:81]: Ok, well once you have run it by him i'm happy with the PR. Given his context, i'd like to wait for that.

**Review by askoufis — APPROVED** [review]

**Review by michaeltaranto — APPROVED** [review]

**Pull request merged**